packages/plugin-ext/src/common/plugin-protocol.ts

```diff
@@ -462,6 +462,7 @@ export interface ViewContainer {
 export interface View {
     id: string;
     name: string;
+    urlBase: string;
```

Comment:
Could you elaborate why do we need it? In other cases we translate icon path to url and transfer it, so code encapsulated in one place. Is there a good reason to do it differently here?

Reply:
The iconUrl is recieved from a code the vscode-extension developer writes. In MSFT examples, the set it relative to the file in which the code is in. See here https://github.com/ariel-bentu/theia-vscode-gaps/blob/master/src/BugsTree.ts - line 45.
or here https://github.com/Microsoft/vscode-extension-samples/blob/master/tree-view-sample/src/nodeDependencies.ts - line 103.

this comes much after we parse the package.json of the extension... so we need to keep context of which view from which extension so when we get it, we can translate it.

Reply:
Does it mean that we pass absolute fs paths to the client and then try to apply them to this url base? I'm a bit reluctant to reveal extension location to clients, but maybe there is no other way.

BTW could you help me to understand why you check for `/extensions/` in the absolute path?

Reply:
> Does it mean that we pass absolute fs paths to the client
No, the extension developer already has the absolute url using the variable __filename. so the paths to icons are absolute. we need to convert them to rest call to the plugin host, so we need the plugin url (hence the baseUrl) and yesterday I added the absolute base of the extension, so the calculation is robust and does not rely on /extension/ in the path (which was wrong in case it was not vsix but extracted extension for example). the only one using this meta data is the tree-views-main.ts